internal/controller/controllers/agent_controller.go

```diff
@@ -137,15 +138,36 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 		}
 	} else { // agent is being deleted
 		if funk.ContainsString(agent.GetFinalizers(), AgentFinalizerName) {
+			if _, has_annotation := agent.GetAnnotations()[BMH_FINALIZER_NAME]; has_annotation {
+				// wait for BMH to be deleted
+				if foundBMH, err := r.bmhExists(ctx, agent); err != nil || foundBMH {
+					if err != nil {
+						log.WithError(err).Warnf("failed to determine if BMH exists for agent")
+					}
+					log.Info("waiting for BMH to be deleted")
+					return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, err
+				}
+				// delete spoke node
+				spokeClient, err := r.spokeKubeClient(ctx, agent.Spec.ClusterDeploymentName)
+				if err != nil {
+					log.WithError(err).Error("failed to create spoke client, node will not be removed")
```

Comment:
Yeah, this is a tough one because the agent and infraenv are being removed there's nowhere to report this kind of thing. It also didn't seem like a recoverable error and we don't really have a way to do multiple retries in something like this without adding to the status.